### ISSUE TITLE
Unable to send images using the sendFile() 

### ISSUE THREAD

**Issue by Roopa77** [pubnub/dart#17]
pubnub.files.sendFile(this.currentChannel, filename, _image.readAsBytesSync()).then((value){
          print(value);
});

I am sending the image file in the above way but i am getting the following error

Unhandled Exception: PubNubException: Request to third party service failed. Details: Query String Parameters not allowed on POST requests.

**Comment by mohanve7** [comment]
Is there any update on this?

**Comment by are** [comment]
Hi, the fix will be released by the end of this week.

**Comment by are** [comment]
Hi there! I've released a RC version, can you please test it out and let me know if this fixes your issue?

To include it, change your `pubnub` dependency in `pubspec.yaml` to this:

```yaml
  pubnub:
    git:
      url: git@github.com:pubnub/dart.git
      ref: v3.0.0-rc
```

Thanks for your cooperation!

**Comment by mohanve7** [comment]
Hi, thanks for the update... Image upload is working fine but plugin throwing a new error in subscription.dart file

 subs.me = pubnub.subscribe(channels: {'${keyset.uuid}.*'}, withPresence: true);
`
[ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: type 'EfficientLengthMappedIterable<String, String>' is not a subtype of type 'Set<String>'
E/flutter (21434): #0      Subscription.presenceChannels 
package:pubnub/…/subscribe/subscription.dart:37
E/flutter (21434): #1      Subscription.presence.<anonymous closure> 
package:pubnub/…/subscribe/subscription.dart:55
E/flutter (21434): #2      _WhereStream._handleData  (dart:async/stream_pipe.dart:193:24)
E/flutter (21434): #3      _ForwardingStreamSubscription._handleData  (dart:async/stream_pipe.dart:157:13)
E/flutter (21434): #4      _rootRunUnary  (dart:async/zone.dart:1198:47)
E/flutter (21434): #5      _CustomZone.runUnary  (dart:async/zone.dart:1100:19)
E/flutter (21434): #6      _CustomZone.runUnaryGuarded  (dart:async/zone.dart:1005:7)
E/flutter (21434): #7      _BufferingStreamSubscription._sendData  (dart:async/stream_impl.dart:357:11)
E/flutter (21434): #8      _DelayedData.perform  (dart:async/stream_impl.dart:611:14)
E/flutter (21434): #9      _StreamImplEvents.handleNext  (dart:async/stream_impl.dart:730:11)
E/flutter (21434): #10     _PendingEvents.schedule.<anonymous closure>  (dart:async/stream_impl.dart:687:7)
E/flutter (21434): #11     _rootRun  (dart:async/zone.dart:1182:47)
E/flutter (21434): #12     _CustomZone.run  (dart:async/zone.dart:1093:19)
E/flutter (21434): #13     _CustomZone.runGuarded  (dart:async/zone.dart:997:7)
E/flutter (21434): #14     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:1037:23)
E/flutter (21434): #15     _rootRun  (dart:async/zone.dart:1190:13)
E/flutter (21434): #16     _CustomZone.run  (dart:async/zone.dart:1093:19)
E/flutter (21434): #17     _CustomZone.runGuarded  (dart:async/zone.dart:997:7)
E/flutter (21434): #18     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:1037:23)
E/flutter (21434): #19     _microtaskLoop  (dart:async/schedule_microtask.dart:41:21)
E/flutter (21434): #20     _startMicrotaskLoop  (dart:async/schedule_microtask.dart:50:5)

`

**Comment by are** [comment]
Hi there! I've uploaded a fix, can you try updating the `pubnub` dependency version to `3.0.0-rc.1` and let me know if its fixed for you? Thanks!

**Comment by mohanve7** [comment]
Thanks! its working

**Comment by are** [comment]
Fixed in [v3.0.0](https://github.com/pubnub/dart/releases/tag/v3.0.0).
